rules_python/whl.py

```diff
@@ -56,21 +57,25 @@ def _dist_info(self):
     return '{}-{}.dist-info'.format(self.distribution(), self.version())
 
   def metadata(self):
-    # Extract the structured data from metadata.json in the WHL's dist-info
-    # directory.
-    with zipfile.ZipFile(self.path(), 'r') as whl:
-      # first check for metadata.json
-      try:
-        with whl.open(self._dist_info() + '/metadata.json') as f:
-          return json.loads(f.read().decode("utf-8"))
-      except KeyError:
-          pass
-      # fall back to METADATA file (https://www.python.org/dev/peps/pep-0427/)
-      with whl.open(self._dist_info() + '/METADATA') as f:
-        return self._parse_metadata(f.read().decode("utf-8"))
+    if self._metadata == None:
+      # Extract the structured data from METADATA in the WHL's dist-info
+      # directory.
+      with zipfile.ZipFile(self.path(), 'r') as whl:
+        with whl.open(self._dist_info() + '/METADATA') as f:
+          # Why are we using email.parser?
```

Comment:
I was really skeptical of this, but the Pip source does the same thing.  Weird.

https://github.com/pypa/pip/blob/b7ed0445258bb18d8d2815e6208f9c996b5f347c/src/pip/_internal/utils/wheel.py#L180

Reply:
That's why I left the comment 😄.  I went through the same level of skepticism.